### PR TITLE
Return all canonical uris when expanding a canonical uri

### PR DESCRIPTION
**Pull request by chdorner**
_tldr;_ This would let us kill two birds with one stone, it allows us to surface old and new annotations 1) when the canonical uri changes, 2) a proposed fix for #3236 

_Long version:_
Before we always only returned the passed-in uri when it ended up being
a canonical uri for the stored document. This works well since all
annotations will have the canonical uri as the `target[source]`.

The only issue is that when a website changes its canonical uri, then we
can't surface the annotations that were made previously because they
won't have the same canonical uri.

This commit fixes that behavior in returning all canonical uris when we
detect that the passed-in uri is a canonical uri.

### CODE REVIEW

**Comment by nickstenning** [comment]
Seems reasonable, but it would still require us to be told about the association between old and new canonical URIs. In this case I guess you've fixed that manually...

**Comment by judell** [comment]
"it would still require us to be told about the association between old and new canonical URIs. In this case I guess you've fixed that manually..."

@chdorner can you unpack that bit for me? In the WP case we currently have:

<link rel="canonical" href="https://blog.jonudell.net/2016/04/02/owning-and-sharing-your-words/" />

I presume this was formerly the http flavor. Since we had that one, then added the https one, didn't we already have both in hand? In other words I'm wondering if this change obviated the need for the manual fixes you made.

**Comment by chdorner** [comment]
Both parts were necessary to fix the WP case.

Quick definition of words for the next paragraph:
* _old_ annotation means an annotation created before the site switched its canonical URL
* _new_ annotation means an annotation created after the site switched its canonical URL

The code fix here is so that we can surface both the _old_ and _new_ annotations on a website that changed its canonical URL. But this only works when we know that two documents are the same, which was the case for this WP.com because both old and new annotations shared at least one document link (`rel-shortlink`).

Me manually adding the https version of http WP.com documents made sure that we can surface all annotations on documents that did not get a new annotation after the https switch.

**Comment by nickstenning** [comment]
So, yeah. This is not the right thing to do. Sorry, I didn't think this through well enough when I reviewed it. See #3249, which is caused by this change.
